lib/util/make-knex.js

```diff
@@ -36,17 +36,32 @@ function initContext(knexFn) {
     // If container is provided, returns a promise for when the transaction is resolved.
     // If container is not provided, returns a promise with a transaction that is resolved
     // when transaction is ready to be used.
-    transaction(container, config) {
-      const trx = this.client.transaction(container, config);
-      trx.userParams = this.userParams;
+    transaction(container, _config) {
+      // Shallow copy _config to avoid side-effects
+      const config = Object.assign({}, _config);
+
+      // Backwards-compatibility: the default value of `config.doNotRejectOnRollback`
+      // changes depending upon whether or not a `container` was provided.
+      if(isUndefined(config.doNotRejectOnRollback)) {
+        config.doNotRejectOnRollback = !container;
+      }
 
-      if (container) {
+      if(container) {
+        const trx = this.client.transaction(container, config);
+        trx.userParams = this.userParams;
```

Comment:
This line seems suspicious to me.  Is there a reason that `userParams` cannot be passed along via the `config` object?